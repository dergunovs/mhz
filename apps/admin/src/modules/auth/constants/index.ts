export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
