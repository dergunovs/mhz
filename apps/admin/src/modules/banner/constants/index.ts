export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
