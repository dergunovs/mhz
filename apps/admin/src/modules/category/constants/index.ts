export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
