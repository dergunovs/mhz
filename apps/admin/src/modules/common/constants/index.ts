export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
