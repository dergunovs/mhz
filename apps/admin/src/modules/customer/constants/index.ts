export const URL_CUSTOMER = '/customers';

export const API_CUSTOMER = 'customer';
