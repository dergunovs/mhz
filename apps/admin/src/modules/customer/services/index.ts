import { Ref, ComputedRef } from 'vue';

import { api, useQuery, IPageQuery, convertParams } from 'mhz-helpers';
import { API_CUSTOMER, ICustomer } from 'mhz-contracts';

export function getCustomers(query: Ref<IPageQuery | number>) {
  async function fn() {
    const params = convertParams(query);

    const { data } = await api.get<{ data: ICustomer[]; total: number }>(API_CUSTOMER, { params });

    return data;
  }

  return useQuery({ queryKey: [API_CUSTOMER, query], queryFn: fn });
}

export function getCustomer(id?: ComputedRef<string | string[]>) {
  async function fn() {
    if (!id?.value) return null;

    const { data } = await api.get<{ data: ICustomer }>(`${API_CUSTOMER}/${id.value}`);

    return data.data;
  }

  return useQuery({ queryKey: [API_CUSTOMER, id], queryFn: fn });
}
