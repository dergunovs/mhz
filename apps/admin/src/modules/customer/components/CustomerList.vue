<template>
  <UiTable
    :headers="tableHeaders"
    :isLoading="!props.customers?.length"
    :modelValue="props.modelValue"
    @update:modelValue="(value) => emit('update:modelValue', value)"
    @reset="(value) => emit('reset', value)"
  >
    <template v-if="props.customers?.length">
      <tr v-for="customer in props.customers" :key="customer._id" data-test="customer-list-row">
        <td data-grow>
          <RouterLink :to="`${URL_CUSTOMER}/${customer._id}`" data-test="customer-list-link">
            {{ customer.email }}
          </RouterLink>
        </td>
        <td data-no-wrap>{{ customer.firstName }} {{ customer.lastName }}</td>
        <td data-no-wrap>
          {{ formatDate(customer.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(customer.dateUpdated) }}
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { ICustomer } from 'mhz-contracts';
import { UiTable } from 'mhz-ui';
import { formatDate, ISortOption } from 'mhz-helpers';

import { URL_CUSTOMER } from '@/customer/constants';

interface IProps {
  customers?: ICustomer[];
  modelValue?: ISortOption;
}

const props = defineProps<IProps>();
const emit = defineEmits<{ 'update:modelValue': [value: ISortOption]; reset: [value: string] }>();

const tableHeaders = [
  { value: 'email', title: 'Email' },
  { value: 'lastName', title: 'Name' },
  { value: 'dateCreated', title: 'Created' },
  { value: 'dateUpdated', title: 'Updated' },
];
</script>
