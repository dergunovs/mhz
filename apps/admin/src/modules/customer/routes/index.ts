import { URL_CUSTOMER } from '@/customer/constants';

export const customerRoutes = [
  {
    path: URL_CUSTOMER,
    name: 'CustomerList',
    component: () => import('@/customer/pages/CustomerListPage.vue'),
  },
  {
    path: `${URL_CUSTOMER}/:id`,
    name: 'Customer',
    component: () => import('@/customer/pages/CustomerPage.vue'),
  },
];
