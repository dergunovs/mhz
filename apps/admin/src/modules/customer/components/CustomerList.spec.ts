import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import CustomerList from './CustomerList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_CUSTOMER } from '@/customer/constants';
import { CUSTOMERS } from '@/customer/fixtures';

const customerListRow = dataTest('customer-list-row');
const customerListLink = dataTest('customer-list-link');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(CustomerList, {
    props: {
      customers: CUSTOMERS.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('CustomerList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(CustomerList)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows rows if customers props', async () => {
    expect(wrapper.findAll(customerListRow).length).toEqual(CUSTOMERS.data.length);

    await wrapper.setProps({ customers: [] });

    expect(wrapper.find(customerListRow).exists()).toBe(false);
  });

  it('generated link to customer', async () => {
    const LINK = `${URL_CUSTOMER}/${CUSTOMERS.data[0]._id}`;

    expect(wrapper.find(customerListLink).attributes('to')).toEqual(LINK);
  });

  it('shows customer content in cells', async () => {
    expect(wrapper.find(customerListLink).text()).toEqual(CUSTOMERS.data[0].email);
  });
});
