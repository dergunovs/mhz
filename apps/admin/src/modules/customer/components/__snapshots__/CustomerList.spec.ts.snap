// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomerList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object]" isloading="false" lang="en">
  <tbody>
    <tr data-test="customer-list-row">
      <td data-grow=""><a to="/customers/64fdd1910e0355e88d03bdb1" data-test="customer-list-link">a@b.ru</a></td>
      <td data-no-wrap="">Alexandr Ivanot</td>
      <td data-no-wrap="">10 сент. 2023 г.</td>
      <td data-no-wrap="">11 сент. 2023 г.</td>
    </tr>
    <tr data-test="customer-list-row">
      <td data-grow=""><a to="/customers/64fdd1910e0355e88d03bdb7" data-test="customer-list-link">test@mail.ru</a></td>
      <td data-no-wrap="">Alex Petrov</td>
      <td data-no-wrap="">11 февр. 2023 г.</td>
      <td data-no-wrap="">12 февр. 2023 г.</td>
    </tr>
  </tbody>
</table>"
`;
